hack/local-up-cluster.sh

```diff
@@ -140,10 +145,40 @@ else
     echo "skipped the build."
 fi
 
+
+function test_etcd {
+  etcd_path=$(which etcd)
+  if [ -z "$etcd_path" ]
+  then
+     echo
+     echo "Failed to find 'etcd' in your PATH, please fix and retry."
+     exit 1
+  fi
+}
+
 function test_docker {
     ${DOCKER[@]} ps 2> /dev/null 1> /dev/null
     if [ "$?" != "0" ]; then
-      echo "Failed to successfully run 'docker ps', please verify that docker is installed and \$DOCKER_HOST is set correctly."
+      echo
+      cat <<'EOF' >&2
+Failed to successfully run 'docker ps', please verify that docker is installed and \$DOCKER_HOST is set correctly. 
```

Comment:
it looks like you took this from `kube::build::ensure_docker_daemon_connectivity` in `build-tools/common.sh`.

I wonder if it might make more sense to refactor that function into `hack/lib/util.sh`, and then call that from both `build-tools/common.sh` and here.